webapp-django/crashstats/signature/views.py

```diff
@@ -499,6 +500,55 @@ def signature_summary(request, params):
         uptimes = sorted(uptimes, key=lambda x: x['count'], reverse=True)
         data['uptimes'] = uptimes
 
+    # Augment graphics adapter with data from another service.
+    if 'adapter_vendor_id' in facets:
+        vendor_hexes = set()
+        adapter_hexes = set()
+        for vendor in facets['adapter_vendor_id']:
+            vendor_hexes.add(vendor['term'])
+            for adapter in vendor['facets']['adapter_device_id']:
+                adapter_hexes.add(adapter['term'])
+        devices_api = models.GraphicsDevices()
+        try:
+            devices_results = devices_api.get(
+                vendor_hex=vendor_hexes,
+                adapter_hex=adapter_hexes,
+            )
+        except models.BadStatusCodeError as e:
+            # We need to return the error message in some HTML form for jQuery
+            # to pick it up.
```

Comment:
Oh this stinks so much. We need to fix that someday. The client didn't do anything wrong. It's our server error that that data couldn't be looked up.